lib/cc/analyzer/issue_validator.rb

```diff
@@ -0,0 +1,49 @@
+module CC
+  module Analyzer
+    class IssueValidator
+      CHECKS = [
+        IssueCategoryValidation,
+        IssueCheckNamePresenceValidation,
+        IssueDescriptionPresenceValidation,
+        IssuePathExistenceValidation,
+        IssuePathPresenceValidation,
+        IssueRelativePathValidation,
+        IssueTypeValidation,
+      ].freeze
+
+      attr_reader :error
+
+      def initialize(issue)
+        @issue = issue
+      end
+
+      def valid?
+        @error.blank?
+      end
+
+      def validate
+        if issue && invalid_messages.any?
+          @error = {
+            message: "#{invalid_messages.join("; ")}: `#{issue}`.",
+            issue: issue,
+          }
+          false
+        else
+          true
```

Comment:
I think there's a not-great edge case here for issues that are unparseable: the `IssueValidator` get's initialized with `nil`, which then hits this branch because `issue && invalid_messages.any?` also evaluates to `nil`. Is the unparseable case still covered somewhere that I'm just not seeing in the diff?